### PR TITLE
treewide: remove various packages from nativeBuildInputs

### PR DESCRIPTION
**Pull request by siraben**
###### Description of changes
These are already included in `stdenv` and don't need to be put in `nativeBuildInputs` again.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Fixed mangled rebase.

File: pkgs/development/compilers/go/1.17.nix [hunk 1]
```diff
@@ -63,7 +63,7 @@ stdenv.mkDerivation rec {
   };
 
   # perl is used for testing go vet
-  nativeBuildInputs = [ perl which pkg-config patch procps ];
+  nativeBuildInputs = [ perl which pkg-config procps ];
```
Comment (zowoq) [inline pkgs/development/compilers/go/1.17.nix:66]:
```diff
diff --git a/pkgs/development/compilers/go/1.17.nix b/pkgs/development/compilers/go/1.17.nix
index 69537dc899e..2adfcd34f1e 100644
--- a/pkgs/development/compilers/go/1.17.nix
+++ b/pkgs/development/compilers/go/1.17.nix
@@ -7,7 +7,6 @@
 , perl
 , which
 , pkg-config
-, patch
 , procps
 , pcre
 , cacert
```
Same should be done for all of the `go` derivations.

**Comment by siraben** [comment]
@zowoq I'll go through all the files I changed with https://github.com/astro/deadnix and squash into their respective commits.

**Comment by zowoq** [comment]
> I'll go through all the files I changed with https://github.com/astro/deadnix and squash into their respective commits.

Seems `deadnix` doesn't remove `patch`? 

I've opened a PR (https://github.com/NixOS/nixpkgs/pull/167088) to remove it anyway so 9bf3653cdd3d4c7d2171ba96120a4af6f5c0f94e can be dropped from this PR.

**Comment by siraben** [comment]
Yeah, I just ran into the issue where `deadnix` doesn't remove `patch`, strange. I'll drop the commit for `patch` in this PR.

**Comment by dasJ** [comment]
@siraben please see #166605 for the next release to staging.

If you experienced weird notifications, I initially pinged you from that issue, but I figured it looks more rude and like I want to shame you publicly.
This was just to inform you about the correct process for the next time and was not intended as an offense.

**Comment by siraben** [comment]
@dasJ thanks, I'll go through that procedure next time.

**Comment by siraben** [comment]
Removed commit that removed bzip2, since it was causing an error in ofborg.

**Comment by Artturin** [comment]
Error: Function called without required argument "python" at /home/artturin/.cache/nixpkgs-review/pr-166605-4/nixpkgs/pkgs/applications/video/davinci-resolve/default.nix:19, did you mean "jython", "python2" or "python3"?

**Comment by siraben** [comment]
@Artturin looks like this is due to python being removed in nixpkgs (instead we need `python2` there)

**Comment by mweinelt** [comment]
`python` is very much a thing, pointing to python2 for legacy reasons. It's probably erroring out due to it being an alias.

File: pkgs/development/libraries/qt-5/modules/qtwebengine.nix [hunk 2]
```diff
@@ -29,7 +29,7 @@ qtModule {
   pname = "qtwebengine";
   qtInputs = [ qtdeclarative qtquickcontrols qtlocation qtwebchannel ];
   nativeBuildInputs = [
-    bison coreutils flex git gperf ninja pkg-config python2 which gn nodejs
```
Comment (Artturin) [inline pkgs/development/libraries/qt-5/modules/qtwebengine.nix:32]: if this is done in a separate PR then this PR won't have to target staging

**Comment by Artturin** [comment]
Result of `nixpkgs-review pr 166605` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>plasma5Packages.akregator</li>
    <li>sway</li>
  </ul>
</details>

File: pkgs/applications/graphics/sane/backends/brscan5/default.nix [hunk 3]
```diff
@@ -27,7 +27,7 @@ stdenv.mkDerivation rec {
     tar xfv data.tar.xz
   '';
 
-  nativeBuildInputs = [ makeWrapper patchelf coreutils ];
+  nativeBuildInputs = [ makeWrapper patchelf ];
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/sane/backends/brscan5/default.nix:30]: Isn't patchelf also part of the stdenv?

Comment (siraben) [reply pkgs/applications/graphics/sane/backends/brscan5/default.nix:30]: Only on linux stdenv, not darwin.

Comment (SuperSandro2000) [reply pkgs/applications/graphics/sane/backends/brscan5/default.nix:30]: Darwin isn't using elf objects. So you shouldn't require it there or am I missing something?

**Comment by Artturin** [comment]
lets merge this before it rots
